src/assets/VueOptionVsCompositionAPI/options-vs-composition-api.md

```diff
@@ -0,0 +1,188 @@
+For the demo application, in `Voting Details` I have defined a component using `Options API` as:
+
+```js
+<script>
+  export default {
+    data() {
+      return {
+        name: 'Sagar Gurung',
+        age: 18,
+        bornYear: 2005,
+        votingStatus: 'Eligible!'
+      }
+    },
+    methods: {
+      increaseAge() {
+        this.age++
+      },
+      decreaseAge() {
+        this.age--
+      }
+    },
+    computed: {
+      getVotingStatus() {
+        return (this.age >= 18) ? 'Eligible' : 'Not Eligible'
+      }
+    },
+    watch: {
+      age(oldAge, newAge) {
+        if(oldAge > newAge) {
+          this.bornYear--
+        } else {
+          this.bornYear++
+        }
+      }
+    }
+  }
+</script>
+```
+In this demo app the properties `data`, `methods`, `computed`, and `watch` have been defined using `Options API`. I will explain the meaning of each of them in the `Composition API` section since we are converting component defined in `Options API` with `Composition API`.
+
+## Converting to `Composition API`
+
+1. ### Defining `data`
+    When defining the data necessary for a component, in `Composition API`, we have to make it reactive to mutate or change values with `ref` or `reactive` provided by Vue.js.
+    So data property can be defined as:
+    ```js
+   // data
+    const data = reactive({
+      name: 'Sagar Gurung',
+      age: 18,
+      votingStatus: "Eligible",
+      bornYear: 2005
+    })
+   ```
+
+2. ### Defining `methods`
+    Defining the methods is very simple. It is just as easy as defining functions in JavaScript.
+    ```js
+    // methods
+    function increaseAge() {
+      data.age++
+    }
+
+    function decreaseAge() {
+      data.age--
+    }
+   ```
+   The above function changes the `age` property of the data defined in `data` which is reactive.
+
+3. ### Defining `computed`
+   To define the computed property Vue provides the method called `computed` through which we can explicitly define the computed function.
+    ```js
+   //computed
+    const getVotingStatus = computed(() => {
+      return (data.age >= 18) ? 'Eligible' : 'Not Eligible'
+    })
+   ```
+   Here the operation `getVotingStatus` returns the voting status based on value of age.
+
+4. ### Defining `watch`
+    Similarly, Vue has provided a `watch` function to watch the change in any data properties defined.
+    ```js
+   //watcher
+    watch(mainData.age, (oldAge, newAge) => {
+      if(oldAge > newAge) {
+        data.bornYear--
+      } else {
+        data.bornYear++
+      }
+    })
+   ```
+   We have watched data `age` in such a way that when age is increased the data `bornYear` will decrease and vice-versa.
+
+The whole script code snippet looks like this at last for the `Composition API`:
+```js
+import { reactive, toRefs, computed , watch} from "vue";
+export default {
+  setup() {
+    // data
+    const data = reactive({
+      name: 'Sagar Gurung',
+      age: 26,
+      votingStatus: "Not Eligible",
```

Comment:
Do you set on purpose a wrong voting status in your initial data? Technically it doesn't matter since getVotingStatus will overwrite this anyway, but I find it a bit confusing...

Reply:
oh sorry i updated the above but forgot the below one. I have adjusted the age itself.